File: algorithm/杭电ACM/ambition0109/2504.cpp
////////////////////System Comment////////////////////
////Welcome to Hangzhou Dianzi University Online Judge
////http://acm.hdu.edu.cn
//////////////////////////////////////////////////////
////Username: ambition0109
////Nickname: Ambition
////Run ID: 
////Submit time: 2010-02-12 18:50:43
////Compiler: Visual C
//////////////////////////////////////////////////////
////Problem ID: 2504
////Problem Title: 
////Run result: Accept
////Run time:0MS
////Run memory:204KB
//////////////////System Comment End//////////////////
#include<stdio.h>
main()
{
	int n,m,x,i,j,t,a,b,k;
	for(scanf("%d",&n);n>0;n--)
	{
		scanf("%d%d",&a,&b);
		for(i=2;;i++)
		{
			t=b*i;
			m=(a<t?a:t);
			for(x=m,j=1;m>1;j++)
			{
				x=m/j;
				if(!((a%x)||(t%x)))
					break;
			}
			if(x==b)
				break;
		}
		printf("%d\n",t);
	}
}

File: algorithm/杭电ACM/ambition0109/2504.java
////////////////////System Comment////////////////////
////Welcome to Hangzhou Dianzi University Online Judge
////http://acm.hdu.edu.cn
//////////////////////////////////////////////////////
////Username: ambition0109
////Nickname: Ambition
////Run ID: 
////Submit time: 2011-04-11 21:00:19
////Compiler: Java
//////////////////////////////////////////////////////
////Problem ID: 2504
////Problem Title: 
////Run result: Accept
////Run time:109MS
////Run memory:6256KB
//////////////////System Comment End//////////////////
import java.util.Scanner;


public class Main {
	
	static int GCD(int a,int b){
		if(a<b){
			int t=a; a=b; b=t;
		}
		while(b!=0){
			int t=b;
			b=a%b;
			a=t;
		}
		return a;
	}
	
	public static void main(String args[]){
		Scanner in=new Scanner(System.in);
		int t=in.nextInt();
		while(t--!=0){
			int a=in.nextInt();
			int b=in.nextInt();
			for(int i=b+b;;i+=b){
				if(GCD(i,a)==b){
					System.out.println(i);
					break;
				}
			}
		}
	}
}
